site/content/integrate/apps/interactivity/dynamic-form-example/_index.md

```diff
@@ -0,0 +1,544 @@
+    const actingUser: UserProfile = context.acting_user;
+
+    // Create a public post using the bot token. This requires the `act_as_bot` permission to be in your App manifest.
+    const publicMessage = `@${actingUser.username} created ticket [${ticket.id}: ${summary}](${ticket.url})`;
+    await createPostAsBot(call.context, publicMessage);
+
+    const privateResponseToUser = 'Thanks for making the ticket! Feel free to create more :)';
+    const callResponse: AppCallResponse = {
+        type: 'ok',
+        markdown: privateResponseToUser,
+    }
+
+    res.json(callResponse);
+});
+
+const createPostAsBot = async (context: AppContext, message: string): Promise<Post> => {
+    const botClient = new Client4();
+    botClient.setUrl(context.mattermost_site_url);
+    botClient.setToken(context.bot_access_token);
+
+    const rootID = context.root_post_id || context.post_id;
+
+    const post = {
+        message,
+        channel_id: context,
+        root_id: rootID,
+    } as Post;
+
+    return botClient.createPost(post);
+};
+```
+
+Now we have a functioning form that allows the user to attach a post to a new ticket, along with a reply `Post` made by our bot that links to the new ticket.
+
+### Static select values
+
+This form works for simple use cases, but ticket systems are usually more complicated than this. What if we have some labels the user needs to choose from? We'll call the labels "tags" in this example so as to not get confused with the `label` field for the select options.
+
+If the App needs to provide a fixed set of values, we can use a `static_select` field. Let's add this field to our form's `fields` array:
+
+```ts
+const tagsField: AppField = {
+    name: 'tags',
+    modal_label: 'Tags',
+    type: 'static_select',
+    value: [],
+    multiselect: true, // When multiselect is true, the user can select multiple options. If multiselect is false or omitted, the user can only select one option.
+    options: [
+        {label: 'Sensitive', value: 'sensitive'},
+        {label: 'Urgent', value: 'urgent'},
+        {label: 'Feature Request', value: 'feature_request'},
+    ],
+};
+```
+
+We can then process the selected values in the form submission like so:
+
+```ts
+type CreateTicketFormSubmission = {
+    summary: string;
+    description: string;
+    tags: AppSelectOption[]; // {label: string; value: string}[]
+};
+
+app.post('/forms/create-ticket/submit', async (req, res) => {
+    const call: AppCallRequest = req.body;
+
+    // Extract the submitted form values
+    const {summary, description, tags}: CreateTicketFormSubmission = call.values;
+
+    // Extract selected values from the tags array
+    const tagValues = tags.map(option => option.value);
+
+    // Process request differently based on tags selected
+    const isUrgent = tagValues.includes('urgent');
+    if (isUrgent) {
+        // ...
+    }
+
+    // ... Process the submission as usual
+});
+```
+
+### Autocomplete select values
```

Comment:
The note talks about autocomplete, but the premise of the next paragraph is about dynamism. I know you use the "autocomplete API" concept, but it still feels weird. Either way, 0/5.